### ISSUE TITLE
go_vet 0%, no problems detected

### ISSUE THREAD

**Issue by cbusbey** [gojp/goreportcard#139]
![screen shot 2016-06-20 at 9 12 28 am](https://cloud.githubusercontent.com/assets/572796/16197294/24f709f4-36c7-11e6-86df-ebb3036b808e.png)

Getting a 0% score with go_vet with "No problems detected. Good job!", bringing our grade down to a "C."

https://goreportcard.com/report/github.com/quickfixgo/quickfix

**Comment by hermanschaaf** [comment]
With 682 files it's a pretty large repo, so our tiny cloud server probably timed out while processing the request. We don't handle it very gracefully right now though. Any suggestions on how we might speed it up are also welcome!

**Comment by cbusbey** [comment]
Perhaps https://github.com/gojp/goreportcard/issues/116 would help.  The vast majority of that code is uninteresting, generated code.

**Comment by hermanschaaf** [comment]
#116 has been closed now, but we can't apply that fix to this quickfix repo issue, because there are no comments indicating that the code was autogenerated in the quickfix repo. For our purposes, this falls in the same category as Kubernetes, in that it is just a really large repository, which we can't handle at the moment.

As far as I can see, we have two ways forward right now:

 - allow checks to run for a longer time (similar to Travis CI)
 - significantly speed up the checks

The second option will help up to a point, until someone comes along with an even bigger repo, or we add a new, much slower check (errcheck is an example of this). I'm therefore leaning towards the first option, which would also necessitate some UI surgery.

**Comment by cbusbey** [comment]
@hermanschaaf to be clear, are you saying that if we add the ".generated.go" suffix to our generated files this would solve the problem?   Or that the repo has so many files, ignored or not by goreportcard, that we would still not get past this issue? We can certainly add this suffix if it would help.

**Comment by hermanschaaf** [comment]
@cbusbey Yes, either a ".generated.go" suffix in the name or a comment on the first line that starts with `// generated ...` would allow those files to not be taken into account by Go Report Card.

Doing so will not immediately solve the issue, because many of the checks are still performed on all files, with the final results down to filtered non-generated files. But this will open up a path for us to drastically improve the efficiency of the quickfix case, by not performing the checks on ignored files in the first place.

One quick win, for example, would be to completely ignore directories/packages that only contain generated files. Another option, which I need to think through a bit more to make sure it is safe, is to  delete the ignored files from our local copy of the repo before performing the checks, and then restoring them afterwards with `git reset HEAD`.

In short: yes, adding a suffix would probably enable us to fix this :)

**Comment by cbusbey** [comment]
Fantastic.  As of a couple of minutes ago, our master branch has added the 'generated.go' suffix to all generated files.  This significantly reduces the number of files that should be reported.

**Comment by hermanschaaf** [comment]
Great, thanks! I will experiment with using this to speed up the checks.

**Comment by hermanschaaf** [comment]
@cbusbey Congrats on your A+! https://goreportcard.com/report/github.com/cbusbey/quickfix

![screenshot from 2016-08-17 00 38 47](https://cloud.githubusercontent.com/assets/1121616/17707442/04b91f50-6413-11e6-9e20-f9c47211997a.png)

I added some code to rename skipped files before the checks are run, then rename them back once they are done. Since the checks usually look at individual files and not packages, this should be okay. If anyone runs into issues with this approach however, let me know.

For the time being this really speeds up repos with lots of generated code, so that makes me happy :+1:
